pkg/kubelet/util/pluginwatcher/plugin_watcher.go

```diff
@@ -115,9 +119,23 @@ func (w *Watcher) registerPlugin(socketPath string) error {
 	if err != nil {
 		return fmt.Errorf("failed to get plugin info using RPC GetInfo at socket %s, err: %v", socketPath, err)
 	}
+
+	// make sure plugin name is unique within a given plugin type.
+	if _, exist := w.registeredPlugins[infoResp.Type][infoResp.Name]; exist {
```

Comment:
Technical note: This has a race as `registerPlugin` is started in a goroutine 

Design note:
This will break intended behaviors like plugins trying to re-register (i.e crash / restart) or even updates of plugins.
Additionally, I'm not sure we should have that check, I believe it should be handled by the pluginwatcher consumer (deviceplugin, csi, ...) as they should be the one handling registration information.

Reply:
you are right, there is race here, i removed the registeredPlugin code, and let handlers do the work.